src/reporting/market.se

```diff
@@ -88,14 +88,27 @@ def initialize(reportingWindow: address, endTime: int256, numOutcomes: int256, p
     i = 0
     while(i < numOutcomes):
         self.shareTokens[i] = SHARE_TOKEN_FACTORY.createShareToken(self.controller)
+        self.shareTokens[i].initialize(self, i)
         i += 1
     self.approveSpenders()
+    require(self.getBranch().isContainerForShareToken(denominationToken) or self.controller.lookup('cash') == denominationToken)
+
     # TODO: log short description
     # TODO: log long description
     # TODO: log tags (up to 0-2)
     # TODO: log outcome labels (same number as numOutcomes)
     # TODO: log type (scalar, binary, categorical)
 
+def shouldCollectReportingFees(denominationToken):
```

Comment:
This should not take in a `denominationToken`, instead it should use `self.denominationToken`.